WalletWasabi.Fluent/ViewModels/Dialogs/Authorization/HardwareWalletAuthDialogViewModel.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using WalletWasabi.CoinJoin.Client.Clients.Queuing;
+using WalletWasabi.Fluent.Model;
+using WalletWasabi.Fluent.ViewModels.Dialogs.Base;
+using WalletWasabi.Hwi;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.Dialogs.Authorization
+{
+	[NavigationMetaData(Title = "Hardware Wallet")]
+	public partial class HardwareWalletAuthDialogViewModel : AuthorizationDialogBase
+	{
+		private readonly Wallet _wallet;
+		private readonly TransactionAuthorizationInfo _transactionAuthorizationInfo;
+
+		public HardwareWalletAuthDialogViewModel(Wallet wallet, TransactionAuthorizationInfo transactionAuthorizationInfo)
+		{
+			_wallet = wallet;
+			_transactionAuthorizationInfo = transactionAuthorizationInfo;
+			IsHardwareWallet = _wallet.KeyManager.IsHardwareWallet;
+			WalletIcon = _wallet.KeyManager.Icon;
+		}
+
+		public string? WalletIcon { get; }
+
+		public bool IsHardwareWallet { get; }
+
+		protected override async Task Authorize()
+		{
+			// Dequeue any coin-joining coins.
+			await _wallet.ChaumianClient.DequeueAllCoinsFromMixAsync(DequeueReason.TransactionBuilding);
+
+			// If it's a hardware wallet and still has a private key then it's password.
+			if (IsHardwareWallet && !_transactionAuthorizationInfo.BuildTransactionResult.Signed)
+			{
+				try
+				{
+					var client = new HwiClient(_wallet.Network);
+					using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
+
+					var signedPsbt = await client.SignTxAsync(
+						_wallet.KeyManager.MasterFingerprint!.Value,
+						_transactionAuthorizationInfo.BuildTransactionResult.Psbt,
+						cts.Token);
+
+					_transactionAuthorizationInfo.Transaction = signedPsbt.ExtractSmartTransaction(_transactionAuthorizationInfo.Transaction);
+
+					Close(DialogResultKind.Normal, true);
+				}
+				catch (Exception ex)
+				{
+					await ShowErrorAsync("Hardware wallet", ex.ToUserFriendlyString(), "Wasabi was unable to sign your transaction");
```

Comment:
dont show a dialog from a dialog, return false, and then we show the error message,

Not authorised or somethign.

Reply:
Done, but have to note that thus we lose the detail what was the reason.
i.e.: Instead of "Hardware wallet is not connected" or "Password is incorrect" we can only say "Authorization has failed".